Docs: add operations/caching page (#772)
grafana/tempo: docs/tempo/website/operations/caching.md

---
title: Caching
weight: 6
---

# Caching

> Tempo can use an external cache to improve query performance.
> The supported implementations are [Memcached](https://memcached.org/) and [Redis](https://redis.io/). 
> Caching is used by the queriers and compactors.

Reviewer: ```suggestion
Caching is used by the queriers and compactors.
Tempo can use an external cache to improve query performance.
The supported implementations are [Memcached](https://memcached.org/) and [Redis](https://redis.io/). 
```

> ### Memcached

Reviewer: Add information to this section.

Author: I've added a small introduction text.

#### Connection limit

As a cluster grows in size, the amount of instances of Tempo connecting to the cache servers will also increase.

> By default Memcached has a connection limit of 1024 and if this limit is exceeded new connections will be refused (resulting in HTTP 500 errors).

Reviewer 2: resulting in Tempo HTTP 500 errors?  Is there a log line or metric we could share here that would help people know when they're overwhelming memcached?

Author: Agree, we can share way more information to help out folks :+1:

> As a cluster grows in size, the amount of instances of Tempo connecting to the cache servers will also increase.
> By default Memcached has a connection limit of 1024 and if this limit is exceeded new connections will be refused (resulting in HTTP 500 errors).
> This is resolved by increasing the connection limit of Memcached.

Reviewer: ```suggestion
As a cluster grows in size, the number of instances of Tempo connecting to the cache servers also increases.
By default, Memcached has a connection limit of 1024. If this limit is surpassed new connections are refused (resulting in HTTP 500 errors).
This is resolved by increasing the connection limit of Memcached.
```

---
title: Caching
weight: 6
---

# Caching

> Caching is used by the queriers and compactors.
> Tempo can use an external cache to improve query performance.

Reviewer 3: ```suggestion
Caching is mainly used to improve query performance by storing bloom filters of all backend blocks which are accessed on every query.

Tempo uses an external cache to improve query performance.
```
